src/datasets/load.py

```diff
@@ -111,11 +111,13 @@ def import_main_class(module_path, dataset=True) -> Optional[Union[Type[DatasetB
     # Find the main class in our imported module
     module_main_cls = None
     for name, obj in module.__dict__.items():
-        if isinstance(obj, type) and issubclass(obj, main_cls_type):
+        if inspect.isclass(obj) and issubclass(obj, main_cls_type):
             if inspect.isabstract(obj):
                 continue
             module_main_cls = obj
-            break
+            obj_module = inspect.getmodule(obj)
+            if obj_module is not None and module == obj_module:
+                break
```

Comment:
This change sounds unrelated to the goal of this PR.

Can you remove this change ? If this is something useful, please open a new PR and explain why you think this is an improvement

Reply:
I proposed this change to make it possible to load the `Wmt` class with `load_dataset_builder("path/to/wmt_utils.py")` and not break `load_dataset_builder("wmt<year>")` due to the `Wmt` class no longer being abstract (to support instantiation). And the way we do this is by giving higher precedence to the builder class defined in `module` than to the imported builder classes. 

But feel free to propose a cleaner solution.